lib/mix/tasks/eliver.bump.ex

```diff
@@ -6,36 +6,129 @@ defmodule Mix.Tasks.Eliver.Bump do
     args |> parse_args |> bump
   end
 
-  defp bump(_) do
+  defp bump(args) do
     case check_for_git_problems() do
       {:error, message} ->
         say message, :red
       {:ok} ->
-        {new_version, changelog_entries} = get_changes()
-        if allow_changes?(new_version, changelog_entries) do
-          make_changes(new_version, changelog_entries)
+        do_bump(args)
+    end
+  end
+
+  defp do_bump(args) do
+    case Keyword.get(args, :multi) do
+      true -> bump_multiple()
+      nil -> bump_normal()
+    end
+  end
+
+  defp bump_multiple() do
+    Eliver.Multiple.list_sub_apps()
+    |> case  do
+      {:ok, sub_apps} ->
+        sub_apps = Map.put(sub_apps, :umbrella, ".")
+
+        changes = Enum.map(sub_apps, fn
+
+          {:umbrella, app_path} ->
+            # Note changes re required for the umbrella app
+            {current_version, new_version, changelog_entries} = get_changes(app_path, :normal)
+            {:umbrella, app_path, current_version, new_version, changelog_entries}
+
+          {app, app_path} ->
+
+          say "\n\n=============================================="
+          say "Bump details for #{Atom.to_string(app)}"
+          say "=============================================="
+
+          get_changes(app_path, :multi)
+          |> case  do
+            nil -> nil
+            {current_version, new_version, changelog_entries} ->
+              {app, app_path, current_version, new_version, changelog_entries}
+          end
+
+
+        end)
+        |> Enum.filter(fn(change) -> not is_nil(change) end)
+
+        if allow_changes?(changes) do
+          make_changes(changes)
         end
+
+      {:error, :unknown_app_structure} ->
+        say "Please note that only standard umbrella app directory structures are supported. Please refer to the documentation for details", :red
+    end
```

Comment:
I reckon this could be used with a `with` statement and relying on pattern matching to do the thing.